decisions/backlog/0021-openid-base-auth.md

```diff
@@ -0,0 +1,79 @@
+# 21. OpenID-Connected Based Endpoint Discovery (Proposal #1)
+
+Date: 2023-09-13
+
+## Status
+
+Proposed
+
+## Context
+
+1. PACT members mentioned challenges in integrating the authentication endpoint (Action `Authenticate` in Tech Specs speak) with their **existing** authentication infratructure
+    1. one reason is that these systems are operated in a special way under pathes and endpoints that cannot easily be modified or adapted
+    2. yet another reason is how the Tech Specs specify the path to the `Authenticate` action ; namely, it is hardcoded as follows: `AuthSubPath/auth/token` with `/auth/token` being fixed.
+    3. To exemplify why this specification is an issue: image the URL of an ********existing******** OAuth2 endpoint was `https://some-domain.com/oauth/token` or`https://some-domain.com/token` ; none such endpoints would be conforming to the current tech specs
+2. Additionally, the `Authenticate` Action is the same for V1 and V2 series
+3. Last, but not least: any update to the Action Authenticate’s path would break backwards-compatibility
+
+## Summary
+
+We propose to extend the Authentication flow for the V2 tech specs. This updated flow makes use of the OpenId Connect Discovery V1 spec (https://openid.net/specs/openid-connect-discovery-1_0.html) to dynamically discover a so-called `token` endpoint. 
+
+If successful, a data recipient authenticates through this endpoint instead of the “regular”`Authenticate` endpoint and its static path / URL.
+
+Otherwise, the authentication flow remains the same and a data recipent attempts to retrieve its token through the regular `Authenticate` Action.
+
+This way, a backwards-compatible authentication flow is established. 
+
+By relying on `OpenId Connect` to discover this endpoint, all parties (Host system implementers, solutions providers, etc.) gain more flexibility in how to operate and to maintain their systems.
```

Comment:
Thank you, @19-SD . I see the following advantages:
1. this approach is based on an existing standard and it is already used for a similar use case (authentication between different organizations)
3. this approach allows to evolve/update the `token` (through updating the `openid-configuration` document)  _without_ needing to communicate this update to the different parties affected; i.e. it would be a more declarative approach compared with what we currently have

Reply:
@zeitgeist thanks for you reply! Regarding point 2, has that been raised by any members as a problem?

Reply:
I suspect that point 2 hasn't been raised as a problem, yet. Once solutions are out in the wild and after a time undergo updates and patches, etc...is when these things break.